pandas/tests/dtypes/test_dtypes.py

```diff
@@ -618,3 +618,8 @@ def test_categorical_equality(self, ordered, other, expected):
         c1 = CategoricalDtype(['a', 'b'], ordered)
         result = c1 == other
         assert result == expected
+
+    def test_mixed(self):
```

Comment:
This is going to fail at the moment. `hash_object_array` tries to work on the mixed array first, and if that fails it falls back to `hash_object_array(a.astype(str))`, which will cause a "collision" with the section set of categoricals. Not sure how I'll handle this yet.